### ISSUE TITLE
Blank/Empty screenshot

### ISSUE THREAD

**Issue by cmonfortep** [ReactiveCircus/android-emulator-runner#273]
Hello, 

A part of another project, we were playing around if it's possible to take an screenshot when running some tests. We are having problems with that since all the screenshots we pull are empty/blank.

To reproduce it, it's as simple ase using the following steps in a github action.
```
      - name: Test
        uses: reactiveCircus/android-emulator-runner@v2
        with:
          api-level: 29
          force-avd-creation: false
          arch: x86_64
          target: google_apis
          profile: Nexus 6
          script: adb shell screencap -p /data/local/tmp/Screenshot.png; adb pull /data/local/tmp/Screenshot.png Screenshot.png;

      - name: Upload screenshot
        if: always()
        uses: actions/upload-artifact@v3
        with:
          name: device-screenshot
          path: ./*Screenshot.png
```

Are we missing something? is it expected that screenshots are blank/empty when using emulators inside github actions?

**Comment by mrk-han** [comment]
@cmonfortep By default the emulator is started headlessly with -no-window, this might be the reason screenshots are not showing up. Try explicitly overriding that in your emulator options and let me know how your screenshots are appearing after.

**Comment by cmonfortep** [comment]
Thanks, I've tested it multiple times with different setups and I'm always getting the same emtpy/blank png. 


This is the latest I tested, and no luck.
```
      - name: Android checks
        uses: reactiveCircus/android-emulator-runner@v2
        with:
          api-level: 29
          force-avd-creation: false
          arch: x86_64
          target: google_apis
          profile: Nexus 6
          emulator-options: -gpu swiftshader_indirect -no-snapshot -noaudio -no-boot-anim
          script: adb shell screencap -p /data/local/tmp/Screenshot.png; adb pull /data/local/tmp/Screenshot.png Screenshot.png;
```

**Comment by mrk-han** [comment]
Are you able to attach the verbose log of the run?

**Comment by cmonfortep** [comment]
I'm playing around with a dummy repository I have. You can take a look to [the run here](https://github.com/cmonfortep/LottieTest/actions/runs/3193315826/jobs/5211753098)

**Comment by mrk-han** [comment]
Hmmm, one question is do you think if we did this same setup without using reactivecircus's action, that screenshots would be working fine?

Maybe we can alter your demo to do the same thing without this action handling the emulator startup.

**Comment by cmonfortep** [comment]
I expect they will work as well, any reason why they shouldn't work?

I've also seen people running screenshot tests on headless emulators, I don't think that's an issue. As a reference, this issue (https://github.com/ReactiveCircus/android-emulator-runner/issues/196) is about screenshots, and the emulator options they are using is `-no-window`.

**Comment by mrk-han** [comment]
@cmonfortep Hey, one of my coworkers had this same issue and said they fixed it by setting this:

```
         sourceCompatibility = 1.8
         targetCompatibility = 1.8
 ```
 
 https://stackoverflow.com/questions/57838904/emulators-cant-load-libgles-emulation-driver

**Comment by cmonfortep** [comment]
Thanks @mrk-han. We have that in our app, and on my dummy app for testing.

Let's see if someone could bring more light here about possible issues around screenshots.

**Comment by mrk-han** [comment]
@cmonfortep See: https://github.com/ReactiveCircus/android-emulator-runner/issues/272#issuecomment-1282877350

I don't think this is related to the android-emulator-runner action

**Comment by mrk-han** [comment]
Closing because this does not seem related to the android-emulator-runner action. See: https://github.com/ReactiveCircus/android-emulator-runner/issues/272#issuecomment-1282877350

**Comment by kopax-polyconseil** [comment]
A solution would be appreciated,. check : https://github.com/actions/runner/issues/2287#issuecomment-1350488499

seems like some people succeed to use  runner with android emulator

**Comment by mrk-han** [comment]
@kopax-polyconseil Using `gpu -host` on the Github Hosted macOS Runners shouldn't work. We don't have GPU access

**Comment by zommerfelds** [comment]
I used to be able to take screenshots with this Action, but now they are mostly coming out black.

This might be the root case, in case anyone wants to follow: https://github.com/actions/runner-images/issues/6814

**Comment by vpuonti** [comment]
Hi,

Try updating to latest Canary version of emulator. I've verified that latest Canary version of emulator no longer has the issue of making blank screenshots.  https://github.com/actions/runner-images/issues/6814#issuecomment-1419003014

Also check the Issuetracker issue where I found this fix for more details: https://issuetracker.google.com/issues/259111479#comment3

I'm not familiar with this runner but setting `channel` parameter to `canary` _should_ pull the Canary version of emulator.
